src/Stack/Types/Config.hs

```diff
@@ -365,6 +365,10 @@ data Resolver
   -- ^ Require a specific GHC major version, but otherwise provide no build
   -- plan. Intended for use cases where end user wishes to specify all upstream
   -- dependencies manually, such as using a dependency solver.
+
+  | ResolverCustom !Text
+  -- ^ A custom resolver based on the given URL. This file is assumed to be
```

Comment:
I suppose this means that if you need to change your snapshot, you need to copy/paste and rename? Is there any way to detect if this assumption has been broken?